domain_config/src/DomainConfigOverrider.php

```diff
@@ -176,7 +176,15 @@ protected function initiateContext() {
     $this->languageManager = \Drupal::languageManager();
     $this->language = $this->languageManager->getCurrentLanguage();
     // Get the domain context.
-    $this->domain = $this->domainNegotiator->getActiveDomain(TRUE);
+    $this->setDomain($this->domainNegotiator->getActiveDomain(TRUE));
   }
 
+  /**
```

Comment:
I had to introduce this setter to allow the domain to be switched on AJAX requests